### DISCUSSION TITLE
for instance members whose type results in`Any`, could pyright output a warning?

### DISCUSSION THREAD

**Discussion by prim** [microsoft/pyright#7511] [Q&A]
for example, for the following code, copyright will not consider a type error by default.

```python

class Command:

	def run(self):
		pass

def f(cmd):
	cmd.run_wrong() # error code
```

When I use vscode to edit these codes, these `Any` member access codes will be displayed in color white. 
This is a kind of reminder for me to add type hint.

![image](https://github.com/microsoft/pyright/assets/1023487/f92e3f2d-94e2-4929-b277-158794f1743a)

Once I added the type hint, pyright would be able to catch these type errors.

```text
pyright a2.py
f:\netease\lbs\gitlab\lbs_client\a2.py
  f:\netease\lbs\gitlab\lbs_client\a2.py:8:6 - error: Cannot access member "run_wrong" for type "Command"
    Member "run_wrong" is unknown (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations
```

However, I work on a huge codebase, and I'm trying to add type hints to most of the code so that pyright can find as many errors as possible.
I would like pright to print a warning for code where attribute access type result is None. 
Then I can be reminded to modify the corresponding code.

**Comment by erictraut** [comment]
> I would like pright to print a warning for code where attribute access type result is None.

I'm not sure what you mean by "where attribute access type results in None". The sample you provided does not meet this criteria. The problem with the sample above is that the type of `cmd` is `Unknown` (an implicit `Any`), so all attribute accesses using this value results in another `Unknown`.

I recommend enabling the `reportUnknownParameterType` diagnostic. It will tell you when you have a parameter that is missing an annotation. Once parameters are annotated, pyright can infer most of the remaining type information.

**Comment by prim** [comment]
Thank you very much for your reply. I found `reportUnknownMemberType` to be what I was looking for.
